src/scipp/coords/transform_coords.py

```diff
@@ -181,7 +181,7 @@ def store(x, c):
         if coord.has_event:
             try:
                 store(da.bins, coord.event)
-            except DimensionError:
+            except (DimensionError, VariableError):
```

Comment:
Is there still a case where this throws `DimensionError`?

Reply:
Yes. https://github.com/scipp/scipp/blob/b00a5aa3edba43dfb58abe05d4e60f9bcafd215d/tests/coords/transform_coords_test.py#L669
With `slice(0, 2)` throws
`scipp._scipp.core.DimensionError: Cannot add coord 'y' of dims (event: 4) to DataArray with dims (event: 8)`